### ISSUE TITLE
Revisit Cartography conventions

### ISSUE THREAD

**Issue by loarie** [inaturalist/inaturalist#599]
Currently we represent observations with markers on maps in two ways:
* 'Featured' markers are the set of 30-200 observations that are displayed on the maps as traditional markers, these are usually the most recent in any search. They are indicated by relatively large teardrops
* 'Background' markers are the remaining observations that we are rendering via tiles when zoom levels are close enough to show markers over grid/heatmap style summaries. They are indicated by relatively small circles

We have been using cartography to distinguish several 'variables': quality, type of taxon (ie iconic taxa), and open/obscured geoprivacy in different ways for these 'featured' and 'background' observations and differently on the web and mobile maps. These cartographic conventions would probably be less confusing if we used a consistent set of conventions.

Currently, these variables are distinguished as follows: quality is distinguished on the 'background' markers via border color but not on the 'featured' markers. Type of taxon is consistently represented by color. Open/obscured geoprivacy is represented using teardrops/circles for the 'featured' markers and as circles/hidden for the 'background' markers (ie obscured background markers are not displayed at all)
![screen shot 2015-06-15 at 4 35 37 pm](https://cloud.githubusercontent.com/assets/598026/8173030/a98152a2-137c-11e5-973f-ad2b03b3e986.png)

@joellebel and I discussed the following proposal:

1. Do not attempt to distinguish quality. The border color on the 'background' markers isn't very noticeable or clear. And since we are not conveying this variable currently on 'featured' markers we propose ignoring it for now. Quality *is* represented on infowindows

2. Distinguish recent/background markers...
 * ...by size of marker (as we are doing now)
 *  ...and by presence/absence of iconic taxon icons (ie embed iconic taxon icons in featured but not background markers as in [the Biocaching app](https://github.com/andyjeffrey/BioCaching/blob/master/BioCaching/Images.xcassets/mapannotation_amphibia_white.imageset/mapannotation_amphibia_white%402x.png))
 * ...but make shape consistent (e.g. teardrops for both featured and background markers)

3. Consistently indicate geoprivacy using teardrops to indicate open and circles to indicate obscured. And emphasize geoprivacy on infowindows since the meaning of these shapes isn't very clear

![screen shot 2015-06-15 at 4 35 41 pm](https://cloud.githubusercontent.com/assets/598026/8173028/a78da414-137c-11e5-8c85-b4c2c2e478d2.png)
![screen shot 2015-06-15 at 4 35 45 pm](https://cloud.githubusercontent.com/assets/598026/8173027/a4eb4c70-137c-11e5-9b98-1bd8bbb50a72.png)

**Comment by naturalistcharlie** [comment]
I would like a way to click on obscured 'background' observations. Now if I
want to see one I have to go through several steps to find the observation
which is confusing.
I think teardrops for background obs. might be a bit 'busy'... though I'm
not sure for certain without seeing it.
On a similar note, there are observations that aren't obscured but have
large uncertainty buffers. These still have small icons. It doesn't make
sense to draw the confidence circle around each one but it would be nice to
know if some of the dots I am looking at are only accurate to 5 km or
whatever.  Not sure the best way to do that either though.
Just some thoughts.

On Mon, Jun 15, 2015 at 8:01 PM, Scott Loarie <notifications@github.com>
wrote:

> Currently we represent observations with markers on maps in two ways:
>
>    - 'Featured' markers are the set of 30-200 observations that are
>    displayed on the maps as traditional markers, these are usually the most
>    recent in any search. They are indicated by relatively large teardrops
>    - 'Background' markers are the remaining observations that we are
>    rendering via tiles when zoom levels are close enough to show markers over
>    grid/heatmap style summaries. They are indicated by relatively small circles
>
> We have been using cartography to distinguish several 'variables':
> quality, type of taxon (ie iconic taxa), and open/obscured geoprivacy in
> different ways for these 'featured' and 'background' observations and
> differently on the web and mobile maps. These cartographic conventions
> would probably be less confusing if we used a consistent set of conventions.
>
> Currently, these variables are distinguished as follows: quality is
> distinguished on the 'background' markers via border color but not on the
> 'featured' markers. Type of taxon is consistently represented by color.
> Open/obscured geoprivacy is represented using teardrops/circles for the
> 'featured' markers and as circles/hidden for the 'background' markers (ie
> obscured background markers are not displayed at all)
> [image: screen shot 2015-06-15 at 4 35 37 pm]
> <https://cloud.githubusercontent.com/assets/598026/8173030/a98152a2-137c-11e5-973f-ad2b03b3e986.png>
>
> @joellebel <https://github.com/joellebel> and I discussed the following
> proposal:
>
>    1.
>
>    Do not attempt to distinguish quality. The border color on the
>    'background' markers isn't very noticeable or clear. And since we are not
>    conveying this variable currently on 'featured' markers we propose ignoring
>    it for now. Quality *is* represented on infowindows
>    2.
>
>    Distinguish recent/background markers...
>     - ...by size of marker (as we are doing now)
>       - ...and by presence/absence of iconic taxon icons (ie embed iconic
>       taxon icons in featured but not background markers as in the
>       Biocaching app
>       <https://github.com/andyjeffrey/BioCaching/blob/master/BioCaching/Images.xcassets/mapannotation_amphibia_white.imageset/mapannotation_amphibia_white%402x.png>
>       )
>       - ...but make shape consistent (e.g. teardrops for both featured
>       and background markers)
>     3.
>
>    Consistently indicate geoprivacy using teardrops to indicate open and
>    circles to indicate obscured. And emphasize geoprivacy on infowindows since
>    the meaning of these shapes isn't very clear
>
> [image: screen shot 2015-06-15 at 4 35 41 pm]
> <https://cloud.githubusercontent.com/assets/598026/8173028/a78da414-137c-11e5-8c85-b4c2c2e478d2.png>
> [image: screen shot 2015-06-15 at 4 35 45 pm]
> <https://cloud.githubusercontent.com/assets/598026/8173027/a4eb4c70-137c-11e5-9b98-1bd8bbb50a72.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599>.
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by loarie** [comment]
@naturalistcharlie background markers currently are clickable - see http://www.inaturalist.org/observations/map#11/23.865/-53.376

I agree there's a 'busy' cost, but would be interesting to consider against the benefits of consistency. Zoom in within the US [here](http://ebird.org/ebird/map/amerob?neg=true&env.minX=&env.minY=&env.maxX=&env.maxY=&zh=false&gp=false&ev=Z&mr=1-12&bmo=1&emo=12&yr=all&byr=1900&eyr=2015) to get a feel for what large numbers of teardrops looks like:

**Comment by naturalistcharlie** [comment]
I thought obscured (or auto obscured) background markers were not
clickable. Maybe I'm in error!
I don't love the big fat ebird markers because they do obscure the map, but
I don't strongly dislike them either. I can definitely see the advantage of
consistency too.

On Mon, Jun 15, 2015 at 9:05 PM, Scott Loarie <notifications@github.com>
wrote:

> @naturalistcharlie <https://github.com/naturalistcharlie> background
> markers currently are clickable - see
> http://www.inaturalist.org/observations/map#11/23.865/-53.376
>
> I agree there's a 'busy' cost, but would be interesting to consider
> against the benefits of consistency. Zoom in within the US here
> <http://ebird.org/ebird/map/amerob?neg=true&env.minX=&env.minY=&env.maxX=&env.maxY=&zh=false&gp=false&ev=Z&mr=1-12&bmo=1&emo=12&yr=all&byr=1900&eyr=2015>
> to get a feel for what large numbers of teardrops looks like:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-112248175>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by loarie** [comment]
obscured background markers currently are hidden (so not clickable but also not displayed), proposal is to make them not hidden and circular shaped like current featured markers.
ignore size of eBird markers (proposal is to keep background markers small) but note shape (proposal is to make them tear dropped shaped)

**Comment by naturalistcharlie** [comment]
Yeah, that all sounds good. I suppose a teardrop vs circle doesn't change
the covered area of the dot much!


On Mon, Jun 15, 2015 at 9:14 PM, Scott Loarie <notifications@github.com>
wrote:

> obscured background markers currently are hidden (so not clickable but
> also not displayed), proposal is to make them not hidden and circular
> shaped like current featured markers.
> ignore size of eBird markers (proposal is to keep background markers
> small) but note shape (proposal is to make them tear dropped shaped)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-112249158>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by kueda** [comment]
I think all these recommendations sound good, EXCEPT for ditching border color to indicate quality grade. While I agree it isn't particularly visible, I actually use that a lot when perusing maps to find stuff that needs ID confirmation. Yes I can get at that info by clicking a marker and yes I could do it by explicitly filtering for it, but it's the *serendipitous* discovery of things that need ID help that makes that cartographic choice useful to me.

Maybe a better approach would be different cartography for observations that need ID (https://github.com/inaturalist/inaturalist/issues/576), b/c frankly I also find it frustrating when I click on a point without a white border and find it doesn't have a photo.

**Comment by naturalistcharlie** [comment]
I also like the 'not research grade' indication though I agree it should be
more prominent. What about giving it a more different boundary color, like
red, or black, or ?.  I agree with Ken-Ichi that it would be nice to be
able to tell if something had no photo also. And for that matter I'd love
it if my observations looked different than other people's observations,
but that's just because I have made so many observations, and because I
can't 'verify' my ID of non-RG observations, of course. But that seems
harder to map since each person would see that differently.

On Tue, Jun 16, 2015 at 3:27 PM, Ken-ichi <notifications@github.com> wrote:

> I think all these recommendations sound good, EXCEPT for ditching border
> color to indicate quality grade. While I agree it isn't particularly
> visible, I actually use that a lot when perusing maps to find stuff that
> needs ID confirmation. Yes I can get at that info by clicking a marker and
> yes I could do it by explicitly filtering for it, but it's the
> *serendipitous* discovery of things that need ID help that makes that
> cartographic choice useful to me.
>
> Maybe a better approach would be different cartography for observations
> that need ID (#576 <https://github.com/inaturalist/inaturalist/issues/576>),
> b/c frankly I also find it frustrating when I click on a point without a
> white border and find it doesn't have a photo.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-112537987>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by pleary** [comment]
Here's something to work off of:
<img width="1061" alt="screen shot 2015-07-24 at 1 33 28 pm" src="https://cloud.githubusercontent.com/assets/48566/8880719/b1f5b9aa-3208-11e5-8538-7ab565b38413.png">

In this example all observations, even obscured ones, are bring show. This is representing all observations as markers, unless the observation is obscured in which case it's a slightly transparent circle. Observations that are research grade have a white dot in their markers. In this example, border colors are not used to indicate anything.

**Comment by loarie** [comment]
This is great Patrick. If (a) we're sticking to the display of the first 200 or so observations returned from the search (ie 'Sort') being distinguishable (at least in the short term) and (b) if we want to make 'Quality' distinguishable, then we have 4 variables we want to display: Sort, Category, Geoprivacy, and Quality and 5 styles to work with: Size, Color, Shape, Decoration, Transparency. All we all agreed that these are the functional constraints of what we want to show?

If so then I guess this is getting into design territory so curious to hear what @joellebel thinks. My instinct above was to do the following:
<img width="665" alt="screen shot 2015-07-24 at 11 21 21 am" src="https://cloud.githubusercontent.com/assets/598026/8881625/301be196-31f6-11e5-83c5-164fa2c7526a.png">
But after seeing how messy the Obscured points make the maps, I now think they really do need to be transparent as Patrick them on staging:
<img width="779" alt="screen shot 2015-07-24 at 10 49 32 am" src="https://cloud.githubusercontent.com/assets/598026/8881044/bf0748d2-31f1-11e5-85b1-5340a923308e.png">
Something like this might be better where transparency is applied to the obscured points, and maybe decoration alone is used to distinguish quality (the current plan is to not display 'unverifiable' by default so only special searches will surface those)
<img width="622" alt="screen shot 2015-07-24 at 11 26 32 am" src="https://cloud.githubusercontent.com/assets/598026/8881779/2ddb681a-31f7-11e5-84d8-e4996ca075b4.png">
Lastly, and back into functional territory, since we only have 4 variables, we technically only need 4 styles. So we could abandon shape if the teardrops are to tricky because of the pointer offset or aesthetically less good than circles.

**Comment by naturalistcharlie** [comment]
I like that the obscured observations are visible as such, and that the
research grade is visible! I am not a big fan of those shape of markers
(the teardrop ones) but that's just personal preference and I don't feel
strongly. Does their bottom point line up with the location, or their
center?

On Fri, Jul 24, 2015 at 1:35 PM, Patrick Leary <notifications@github.com>
wrote:

> Here's something to work off of:
> [image: screen shot 2015-07-24 at 1 33 28 pm]
> <https://cloud.githubusercontent.com/assets/48566/8880719/b1f5b9aa-3208-11e5-8538-7ab565b38413.png>
>
> In this example all observations, even obscured ones, are bring show. This
> is representing all observations as markers, unless the observation is
> obscured in which case it's a slightly transparent circle. Observations
> that are research grade have a white dot in their markers. In this example,
> border colors are not used to indicate anything.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-124591262>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont

**Comment by loarie** [comment]
charlie - the plan if we go with the teardrops would be that they would point to the location (rather than be centered on it), but still unclear whether thats possible/worth it

**Comment by naturalistcharlie** [comment]
Is this all referring to all maps or just searches? If all maps I don't
know if I like the idea of not having unverifiable absent from maps. I
sometimes will put something in without a photo or with only a poor photo
in the hope that someone else will go back and get more data - especially
if it's an important observation for understanding the range of the
species, etc. I personally don't think filtering these sorts of things away
is necessarily the best approach - i think it would be a shame if they were
absent from species maps or 'all observations' maps... Maybe I'm wrong...
but just a thought.

I think the reason I don't like the pins is they take up more space than
the circles and block significant areas north of where the observation
occurs. But again not a strong dislike...

On Fri, Jul 24, 2015 at 2:47 PM, Scott Loarie <notifications@github.com>
wrote:

> charlie - the plan if we go with the teardrops would be that they would
> point to the location (rather than be centered on it), but still unclear
> whether thats possible/worth it
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-124623859>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont

**Comment by naturalistcharlie** [comment]
one other thought... i know there is a lot here already... but... what
about some way some day to depict high locational uncertainty.  There are
plenty of reasons to have those sorts of observations but precision mapping
isn't one of them.. when looking at range or habitat it would be nice to be
able to distinguish them, some day. Just another thought

On Fri, Jul 24, 2015 at 2:49 PM, Charlie Hohn <naturalist.charlie@gmail.com>
wrote:

> Is this all referring to all maps or just searches? If all maps I don't
> know if I like the idea of not having unverifiable absent from maps. I
> sometimes will put something in without a photo or with only a poor photo
> in the hope that someone else will go back and get more data - especially
> if it's an important observation for understanding the range of the
> species, etc. I personally don't think filtering these sorts of things away
> is necessarily the best approach - i think it would be a shame if they were
> absent from species maps or 'all observations' maps... Maybe I'm wrong...
> but just a thought.
>
> I think the reason I don't like the pins is they take up more space than
> the circles and block significant areas north of where the observation
> occurs. But again not a strong dislike...
>
> On Fri, Jul 24, 2015 at 2:47 PM, Scott Loarie <notifications@github.com>
> wrote:
>
>> charlie - the plan if we go with the teardrops would be that they would
>> point to the location (rather than be centered on it), but still unclear
>> whether thats possible/worth it
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-124623859>
>> .
>>
>
>
>
> --
> ============================
> Charlie Hohn
> Montpelier, Vermont
>
>


-- 
============================
Charlie Hohn
Montpelier, Vermont

**Comment by loarie** [comment]
So based on what Patrick has on staging at the moment, I think the cartography can be summed up as follows:
<img width="505" alt="screen shot 2015-07-27 at 4 27 25 pm" src="https://cloud.githubusercontent.com/assets/598026/8920300/766ed062-347c-11e5-8248-1473cb3c5837.png">
(1) Patrick, it seems as though you solved the problem of positioning the pointers so they point at (rather than center on) the location? Because it seems that the 'Sort' set of Google Maps teardrop markers are now completely covering the SVG tile ones underneath which is great. It does look like the Google Maps circular markers are not completely covering the SVG tile ones however (see upper left here):
<img width="333" alt="screen shot 2015-07-27 at 4 24 44 pm" src="https://cloud.githubusercontent.com/assets/598026/8920307/7d7b45de-347c-11e5-95d8-e189b4fe3227.png">
(2) Also, the Google Map markers always have white dots. If we are going to use the white dot to indicate research grade its probably worth adding a separate set of Google Map Marker pngs with dots/no dots and displaying them accordingly - is this a lot of work?
(3) Lastly, when we break 'casual' into 'unverifiable' & 'needs ID' with the forthcoming NeedsID changes, are we going to try to indicate those groups with some sort of cartography, if so, would be nice to start thinking of the design since that release is close. e.g.
<img width="604" alt="screen shot 2015-07-27 at 4 27 40 pm" src="https://cloud.githubusercontent.com/assets/598026/8920305/796d539c-347c-11e5-8be9-0fe0be53d78c.png">

**Comment by joellebel** [comment]
Having a larger marker to indicate a recent observation is ok for the time being, but in the future I am interested in exploring different solutions for that. Perhaps, a larger marker isn't necessary if you can hover over an observation snippet and the observation marker summary pops up. 

### Obscured Location
While I understand why the obscured location markers are transparent, I am not sure that's the right effect. Transparent icons/items are typically used to indicate a disabled or incomplete state. My concern is that folks will think those observations are either incomplete, unverifiable or disabled for some reason. I could almost see using a transparency for the "Unverifiable" state. At this point, I don't have a better solution though. 


### (2) White Dots for Large Markers
Ok. Sounds good. @pleary please get back to me if you can use the existing SVGs for this or I need to create new dotless PNGs for the large markers.

### (3) Casual, Unverifiable, Research Quality State
I'm a little concerned that we have a lot of different makers and states going on which will make for quite a lengthy map legend and a lot to digest for both new users. Even an advanced user may not remember all these states every time they return to the map. However, if Unverifiable markers are off by default that will help. 

I also feel Needs ID markers can be turned on and off with a filter. (After all, that's why we're making a variety of filters.) Ken-ichi can you expand more on this... "Yes I can get at that info by clicking a marker and yes I could do it by explicitly filtering for it, but it's the serendipitous discovery of things that need ID help that makes that cartographic choice useful to me." Yes, serendipitous, but can you elaborate on the use cases around why identifiers need to see both a specific needs id marker next to research quality when they can simply turn these items on and off with a filter. For example, if we take identifiers to an Identifier section, can we merely have the research quality observations off by default? I'm not totally opposed, just want to understand the use cases around it.

**Comment by kueda** [comment]
I am an identifier but I also use the rest of the site. One of the things that sort of keeps me stuck on iNat for longer than I'd like is *accidentally* finding things I feel like I need to do, like identifying observations that need it. So...

Serendipity use case 1: I'm looking at someone else's observation (b/c it's cool, b/c I'm identifying it, whatever), and I notice from the map that there are three nearby observations of the same species. I can immediately tell from the cartography that only one of them needs an ID so I click on it, go to its detail page, and add an ID.

Serendipity use case 2: I just learned how to ID something new, so I'm looking at the map for that species. I can immediately see that while most observations have been confirmed, there a couple in the Bay Area that haven't been, so I focus my efforts on them.

Serendipity use case 3 (uber nerdy): I'm looking at a taxon map and notice there's a RG observation in a county that does *not* have this taxon on its checklist. This signals to me that something has gone wrong that I need to investigate.

**Comment by joellebel** [comment]
Is anyone else finding this zoom level particularly difficult to visually parse? It's zoom level 10.

![image](https://cloud.githubusercontent.com/assets/11527994/8945854/0eb09a04-3540-11e5-8c8f-d8b01185bbf0.png)

How's about simply not showing a marker if over 80% of it covered by another? Chances are nobody will click on it anyways or couldn't because the mouse cursor is too big. Of course, the absent makers would appear when one zooms in more.

**Comment by kueda** [comment]
I know it's messy, but it reflects the actual state of the data. If you ignored overlapping markers you might get the false impression that the visible points represented all the data available and would not feel compelled to zoom in and reveal all the other observations that were hidden. The heatmap and the transparent marker treatment we're currently using when more zoomed out achieve this by looking smooshy and abstracted, but if you use the same treatment for the markers and just hide a bunch of them, you're hiding data instead of abstracting it.

**Comment by joellebel** [comment]
I guess this is where groupings helps. ;)

**Comment by loarie** [comment]
To me 99% of the difficulty in visually parsing comes from the obscured data since it's essentially noise in the signal of obs distribution, but if we want obscured obs clickable I can't think of a better solution - making them opaque would almost certainly make things worse though. Another option is not to make the gray circle to colored teardrop style switch until a higher zoom level, but since the appropriate level for that switch depends on obs density it might make the Bay Area less messy but will likely make for a less ideal switch elsewhere

**Comment by joellebel** [comment]
@kueda I hear what you're saying about hiding data, but think there will still be so many markers visible it will warrant a further zoom at which point more markers will be visible.

**Comment by pleary** [comment]
for me, yes that view of that data is hard to parse, and I kind of love it. That's a view of all obs, and there are so many it's difficult to see them all, and that makes me happy to see so much data. I can always zoom in, or do a search and view a filtered map. I'm hoping we'll have an epiphany and come up with an awesome way to show maps and observation search results in real time, and I'm kind of hoping that will come out of https://github.com/inaturalist/design/issues/20.

And FWIW it's worth we're already doing as you suggest joelle and do some data summarization, but it's at a scale that is sort of imperceptible to the user. I wanted just enough points in there to make it look full w/o needing to render every single point, so as to not unnecessarily overload the tileserver.

**Comment by joellebel** [comment]
Okeedoke. I guess I'm outvoted here. Food for thought for the future... If the only thing we're really showing to the user at this zoom level...

![image](https://cloud.githubusercontent.com/assets/11527994/8946689/2fd514ce-3547-11e5-9bd5-8afccd91e006.png)

...Is that there is a lot of data and biodiversity, perhaps there are other ways of doing that more directly. I will tackle this in the future.

**Comment by joellebel** [comment]
@kueda Thanks for the serendipity use cases. I had an inkling #1 and #2 existed. Did not think of #3. Great. 

@loarie Any thoughts on these use cases? Earlier when I mentioned #1 to you, you seemed a bit lukewarm about it. Is this still the case? Also, as an identifier do these use cases resonate with you and are there any other ones?

**Comment by loarie** [comment]
I agree with kenichi on all of that. I was lukewarm on the idea that identifiers would Id a needs Id marker based on looking at rg markers around it but maybe I misunderstood u

**Comment by naturalistcharlie** [comment]
Another thing about unverifiables... Most are obviously of limited value
but some may be important "leads". For instance if someone thinks they saw
a rare species or an invasive but has no photo or a poor photo. Land
managers may still want to go check if it's there.
I use the range maps in the observation page a lot and often do look at
other nearby ones. And every once in a rare while something rg is still
wrong. I find the most questionable ids by looking out of range. 95% of the
time a misid and 5% a neat range extension.

Anyhow this all looks wonderful as does the new obscuring idea. Thank you!!!

On Wednesday, July 29, 2015, Scott Loarie <notifications@github.com> wrote:

> I agree with kenichi on all of that. I was lukewarm on the idea that
> identifiers would Id a needs Id marker based on looking at rg markers
> around it but maybe I misunderstood u
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/599#issuecomment-126027972>
> .
>


-- 
Sent from Gmail Mobile

**Comment by pleary** [comment]
> @loarie : (1) ... It does look like the Google Maps circular markers are not completely covering the SVG tile ones however (see upper left here):

The Google Maps circles are sitting on top of their observations - the center is not lined up with the observation. The ones from the tileserver are definitely on point. We can add an offset to the Google markers to fix that.

> (2) Also, the Google Map markers always have white dots. If we are going to use the white dot to indicate research grade its probably worth adding a separate set of Google Map Marker pngs with dots/no dots and displaying them accordingly - is this a lot of work?

Turns out Google maps can render markers with SVGs. You can customizable color, border, size, and rotation but nothing else. As for the content within the marker - Google doesn't natively support any good custom labels on markers, and all the libraries I tried didn't do what we want. They could create labels, but not tied to and on the same plane as the markers, so it was a mess. Given that the PNG markers would be noticeably different, and we can't replace all old markers, looks like we should bail and create empty versions of the existing ones.

> @joellebel : Having a larger marker to indicate a recent observation is ok for the time being, but in the future I am interested in exploring different solutions for that. Perhaps, a larger marker isn't necessary if you can hover over an observation snippet and the observation marker summary pops up.

I agree it would be great to get off the custom Google map markers. Would it be crazy to remove them entirely and have like a "zoom to this observation" option on the observation snippet which updates the map to show that observation, or simply open it's infowindow. Or maybe clicking on the observation snippet DIV itself does it.

> I'm a little concerned that we have a lot of different makers and states going on which will make for quite a lengthy map legend and a lot to digest for both new users. Even an advanced user may not remember all these states every time they return to the map. However, if Unverifiable markers are off by default that will help.

Definitely agree not to get too complicated. But if we are able to remove the big markers, then we're left with just 3 variables - color, dot or no dot, and shape (all circles and only circles are transparent)

**Comment by joellebel** [comment]
> I agree it would be great to get off the custom Google map markers. Would it be crazy to remove them entirely and have like a "zoom to this observation" option on the observation snippet which updates the map to show that observation, or simply open it's infowindow. Or maybe clicking on the observation snippet DIV itself does it.

I am a little leery of zooming in and out as it can be a bit jarring if you're rapidly clicking or hovering over snippets, but am certainly open to seeing it in action! I also may not be envisioning it correctly.

Here's an example of a model I've been exploring:

http://www.trulia.com/for_sale/Chicago,IL/map_v

...but again, am looking forward to looking at this more holistically in the future. ;)

**Comment by pleary** [comment]
Staging is running all latest changes http://staging.inaturalist.org/observations?d1=2015-08-1

I think I was just massively confused, and we ultimately probably don't need to change much. I was thinking mappable contained conditions about wether or now an observation is obscured, but it's way more basic things than that:
 - can't be private
 - can't be > 10k obscured (10k exactly is OK because we want to show obscured threatened taxa)
 - can't be captive
 - can't have an inaccurate location

So unless we want to start allowing any of those types, mappable doesn't need to change.

To summarize, here are all the changes ready to go:

Node:
 - the tileserver will replace the smaller circles we use now with two styles - a pin-like marker that sits on top of the observation and 'points' to it, and a faded circle for observations with obscured coordinates (the circle surrounds the observation). Anything that is research grade will have a white dot in it
 - observations with obscured coordinates will no longer be hidden at zoom levels higher than 8
 - tileserver uses cartoCSS for much simpler style code

Rails:
 - Large 'featured observation' map markers match styles with the above - obscured obs stay circles (properly surrounding their observation now), and research grade markers contain a dot while non-research grade markers now have no dot
 - updated UTFGrid interaction to be offset from observation location for map tiles using markers that sit on top of their observations now

**Comment by kueda** [comment]
Cool I think this looks good. Not to feature-creep too much, but any thoughts on where we might incorporate a legend? I feel like we could get away with a single button on all taxonmaps that opens a popup explaining what all the symbology means.

**Comment by pleary** [comment]
After a flurry of activity today, we came out with a nice looking legend http://staging.inaturalist.org/observations . It's a little large, so it's only showing on the larger maps on observations/index, observations/map, taxa/show and taxa/map right now.

I was also able to get the Unknown taxon (?) markers showing up in the tileserver using the same icons as we use in the Google JS: http://staging.inaturalist.org/observations/map?#13/37.863/-122.307

If anyone has any recommended changes before deploy, please let me know tonight and I should be able to get to them tomorrow morning. Tomorrow I'll leave instructions for how to deploy this in my absence, otherwise I'll be back starting next Wednesday.

**Comment by kueda** [comment]
I think it's looking great. Looking forward to the instructions.

**Comment by joellebel** [comment]
Looking good. It would be nice for the next release, if clicking outside of the legend dismisses it.

**Comment by kueda** [comment]
Most of this has been merged and deployed. Let's open new issues for remaining issues, e.g. https://github.com/inaturalist/iNaturalistAPI/issues/5
